lib/main.js

```diff
@@ -394,7 +394,7 @@ define(['./errors'], function(errors) {
     // for easy membership testing with the "in" keyword
     function setLike(list, fill /* default true*/) {
         if(fill === undefined) fill = true;
-        var setLike = {};
+        var setLike = Object.create(null);
```

Comment:
I see, thanks for the detailed explanation. As small bit of info, ES6 introduces a replacement for the older (and pretty shitty) `for ... in ...`, using `for ... of ...` which does object key/val iteration properly, running over own props but not prototype props. This has more immediate applications to arrays, since you can now do `for (let val of arr) { if(...) break; ... }` and not worry about `v` becoming prototype keys like `map` or `length`, but it works the same for objects, and does pair extraction too, using `for (let[key,value] of obj) { ... }`